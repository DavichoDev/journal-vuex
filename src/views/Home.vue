<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <h1>Hola mundillo :3</h1>
    <button @click="goToDaybook" class="btn btn-primary">Primary</button>
    <button @click="goToDaybook" class="btn btn-secondary">Secondary</button>
    <button @click="goToDaybook" class="btn btn-success">Success</button>
  </div>
</template>

<script>
export default {
  name: 'Home',
  components: {
  },
  methods: {
    goToDaybook() {
      this.$router.push({name: 'no-entry'})
    }
  }
}
</script>
