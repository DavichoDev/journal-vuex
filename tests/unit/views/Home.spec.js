import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home.vue'

describe('Home.vue tests', () => {

    let wrapper

    beforeEach(() => {
      wrapper = shallowMount( Home )

      jest.clearAllMocks()
    })

    test('Debe de renderizar el componente correctamente', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('Hacer click en un boton debe de redirecciona a no-entry', () => {
        
        const mockRouter = {
            push: jest.fn()
        }

        wrapper = shallowMount(Home,{
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect( mockRouter.push ).toHaveBeenCalled()
        expect( mockRouter.push ).toHaveBeenCalledWith({name: 'no-entry'})

    })

})