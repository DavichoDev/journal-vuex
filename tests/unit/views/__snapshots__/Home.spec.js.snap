// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue tests Debe de renderizar el componente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola mundillo :3</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
